[BUG] Client not able to deal with surrogates/emoji in data

### What is the bug?
When running the neptune to opensearch replicator on AWS, we ran into problems with replicating emojis:

```
[ERROR] UnicodeEncodeError: 'utf-8' codec can't encode characters in position 1999-2000: surrogates not allowed
Traceback (most recent call last):
  File "/opt/python/lambda_function.py", line 183, in lambda_handler
    default_handler_replication(lease, wait_time, execution_end_time)
  File "/opt/python/lambda_function.py", line 131, in default_handler_replication
    raise e
  File "/opt/python/lambda_function.py", line 113, in default_handler_replication
    if not stream_records_processor.write_with_metrics(query_queue, lease, lease_manager, metrics_publisher_client):
  File "/opt/python/stream_records_processor.py", line 162, in write_with_metrics
    result = self.write(query_queue)
  File "/opt/python/stream_records_processor.py", line 144, in write
    return stream_records_handler.write_records(query_queue.get())
  File "/opt/python/replication_handler.py", line 301, in write_records
    raise e # Raise exception to avoid updating checkpoint in case of write failure
  File "/opt/python/replication_handler.py", line 298, in write_records
    future.result() # Need to call result() so that exception can be raised
  File "/var/lang/lib/python3.9/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/var/lang/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/var/lang/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/python/replication_handler.py", line 331, in __run_query__
    raise e
  File "/opt/python/replication_handler.py", line 320, in __run_query__
    self.execute_query(batch_query[0], batch_size)
  File "/var/task/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/var/task/retrying.py", line 206, in call
    return attempt.get(self._wrap_exception)
  File "/var/task/retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/var/task/six.py", line 719, in reraise
    raise value
  File "/var/task/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/var/task/neptune_to_es/neptune_to_es_handler.py", line 465, in execute_query
    success, errors = bulk(self.get_es_client(), actions, max_retries=3, chunk_size=batch_size,
  File "/var/task/opensearchpy/helpers/actions.py", line 411, in bulk
    for ok, item in streaming_bulk(
  File "/var/task/opensearchpy/helpers/actions.py", line 320, in streaming_bulk
    for bulk_data, bulk_actions in _chunk_actions(
  File "/var/task/opensearchpy/helpers/actions.py", line 166, in _chunk_actions
    ret = chunker.feed(action, data)
  File "/var/task/opensearchpy/helpers/actions.py", line 127, in feed
    cur_size += len(data.encode("utf-8")) + 1
```

### How can one reproduce the bug?

Try to replicate a string with emojis: Kan Boer Ayoub tractor 🚜 pulling winnen? | Motor Maniacs

### What is the expected behavior?

Strings with emojis replicate without errors.

### What is your host/environment?

AWS Lambda Python 3.9 x86_64

### Do you have any screenshots?

No

### Do you have any additional context?

This change resolves this particular error while emojis do not seem affected:

```
# opensearchpy/client/actions.py line 126
cur_size += len(data.encode("utf-8")) + 1
# change to
cur_size += len(data.encode("utf-8", errors="surrogatepass")) + 1
```

Setup in accordance with https://docs.aws.amazon.com/neptune/latest/userguide/full-text-search-cfn-setup.html
Repost.aws question: https://repost.aws/questions/QUPblDivgvT-KyoSvutzj5vw/bug-in-aws-neptune-to-open-search-replication-cloud-formation-missing-import-es-helper-in-replication-handler-py-eu-west-1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Deprecated
 ### Removed
 ### Fixed
+- Fix `UnicodeEncodeError` on surrogate/emoji characters in bulk chunk sizing by passing `surrogatepass` error handler ([#1086](https://github.com/opensearch-project/opensearch-py/pull/1086))
 ### Security
 ### Dependencies
 - Refresh `benchmarks/` poetry lock to pick up aiohttp 3.13.5, urllib3 2.6.3, requests 2.33.1, pygments 2.20.0 (clears Mend CVE findings) ([#1046](https://github.com/opensearch-project/opensearch-py/pull/1046))

--- a/opensearchpy/helpers/actions.py
+++ b/opensearchpy/helpers/actions.py
@@ -120,11 +120,12 @@ class _ActionChunker:
         raw_data, raw_action = data, action
         action = self.serializer.dumps(action)
         # +1 to account for the trailing new line character
-        cur_size = len(action.encode("utf-8")) + 1
+        # surrogatepass matches the rest of the client (transport.py, http_requests.py, etc.)
+        cur_size = len(action.encode("utf-8", "surrogatepass")) + 1
 
         if data is not None:
             data = self.serializer.dumps(data)
-            cur_size += len(data.encode("utf-8")) + 1
+            cur_size += len(data.encode("utf-8", "surrogatepass")) + 1
 
         # full chunk, send it and start a new one
         if self.bulk_actions and (

